File: Algorithms/Greedy/Sherlock and The Beast.php
<?php

/*
 * Complete the 'decentNumber' function below.
 *
 * The function accepts INTEGER n as parameter.
 */

function decentNumber($n) {
    # Write your code here
    $numberOfFives = $n - ($n % 3);

    while ($numberOfFives >= 0) {
        if (($n - $numberOfFives) % 5 == 0) {

            echo str_repeat('5', $numberOfFives) . str_repeat('3', $n - $numberOfFives) . "\n";
            return;
        }

        $numberOfFives -= 3;
    }

    echo "-1\n";
}

$t = intval(trim(fgets(STDIN)));

for ($t_itr = 0; $t_itr < $t; $t_itr++) {
    $n = intval(trim(fgets(STDIN)));
    decentNumber($n);
}

File: Algorithms/Greedy/Sherlock and The Beast.py
#
# Complete the 'decentNumber' function below.
#
# The function accepts INTEGER n as parameter.
#

def decentNumber(n):
    # Write your code here
    number_of_fives = n - (n % 3)

    while number_of_fives >= 0:
        if (n - number_of_fives) % 5 == 0:
            print("5" * number_of_fives + "3" * (n - number_of_fives))
            return

        number_of_fives -= 3

    print(-1)

if __name__ == '__main__':
    t = int(input().strip())

    for t_itr in range(t):
        n = int(input().strip())

        decentNumber(n)
